src/Components/firebase-access.jsx

```diff
@@ -0,0 +1,359 @@
+          .where("Sector","==",Sector)
+          .where("SectorPos","<=", cutoff)
+          .orderBy("SectorPos").limit(NumOfSymbols).get()
+      Sectors.forEach(function(doc){
+          symbols.push(doc.data().Symbol)
+      })
+
+  }
+  else{
+
+      let StockInfo =  await db.collection("Ticker-Info").doc("Stock").get();
+      let numOfStocks = StockInfo.data().NumOfStocks - 1;
+      let cutoff = Math.floor((Math.random()  * (numOfStocks - NumOfSymbols))+NumOfSymbols);
+      let Stocks = await db.collection("Ticker-Info").doc("Stock").collection("Stocks")
+          .where("RandomPos",">=", cutoff)
+          .orderBy("RandomPos").limit(NumOfSymbols).get()
+      Stocks.forEach(function(Stock){
+          symbols.push(Stock.data().Symbol)
+      })
+
+  }
+  return symbols
+}
+
+// requests prices and technical indicator images to be written to the database
+export const initializeQuiz = async (symbols, roomId, periodLen, endDates) => {
+  var formData = new FormData();
+  formData.append('symbol',JSON.stringify(symbols));
+  formData.append('RoomId',roomId);
+  formData.append('end-date',JSON.stringify(endDates));
+  try{
+      await fetch('http://localhost:8080/get-prices', {
+          method: 'POST',
+          mode: 'cors',
+          body: formData
+      })
+  }
+  catch(err) {
+      console.log("Error is " +  err)
+  }
+  formData.append('periodLen',periodLen)
+  try{
+      await fetch('http://localhost:8080/get-stock-image', {
+          method: 'POST',
+          mode: 'cors',
+          body: formData
+      })
+  }
+  catch(error){
+      console.log("Error is " +  error)
+  }
+}
+
+// gets a list of all symbols being tracked
+export const getSymbols = async (db, roomID) => {
+  const symbol = await db.collection('Rooms').doc(roomID).get();
+  const symbolData = await symbol.data();
+  return symbolData.symbols;
+}
+
+export const getDate = async (db, roomID) => {
+  const roomDoc = await db.collection('Rooms').doc(roomID).get();
+  const roomData = roomDoc.data();
+  return roomData.dates[roomData.day_index];
+}
+
+// returns current price for all symbols being tracked
+export const getCurrentPrice = async (db, roomID) => {
+  const roomDoc = await db.collection('Rooms').doc(roomID).get();
+  const roomData = roomDoc.data();
+  const dayIndex = roomData.day_index;
+  const symbolNamesArray = roomData.symbols;
+
+  var prices = [];
+  
+  // save all current prices in an array
+  for (var index = 0; index < symbolNamesArray.length; index++) {
+    const currentName = symbolNamesArray[index];
+    const symbolPricesDoc = await db.collection('Rooms').doc(roomID).collection(currentName).doc('Prices').get();
+    const symbolPricesData = symbolPricesDoc.data();
+    
+    const currentPrice = symbolPricesData.prices[dayIndex];
+    prices.push(currentPrice);
+  }
+
+  return prices;
+}
+function compDoc(a, b){
+  if ( a.value < b.value ){
+      return -1;
+  }
+  if ( a.value > b.value ){
+  return 1;
+  }
+  return 0;
```

Comment:
be consistent with spacing and indentation

```suggestion
  if (a.value < b.value) {
      return -1;
  }
  if (a.value > b.value) {
    return 1;
  }
  return 0;
```